integration/testutil_soci.go

```diff
@@ -0,0 +1,72 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"strings"
+
+	shell "github.com/awslabs/soci-snapshotter/util/dockershell"
+	"github.com/containerd/containerd/platforms"
+	"github.com/opencontainers/go-digest"
+)
+
+const (
+	// copied from `soci/soci_index.go` for convenience so we don't always need to
+	// import the `soci` pkg only to use the default values.
+	// default span size (4MiB)
+	defaultSpanSize = int64(1 << 22)
+	// min layer size (10MiB)
+	defaultMinLayerSize = 10 << 20
+)
+
+// buildIndex builds a soci index with default span size and 0 min-layer-size, so every layer
+// will have a corresponding ztoc.
+func buildIndex(sh *shell.Shell, src imageInfo) string {
```

Comment:
I think functional parameters are a more idiomatic way to do this:

```
func buildIndex(sh *shell.Shell, src imageInfo, opts ...buildIndexOpts) string
```

The benefits being:
1) We have 1 name in our codebase
2) We only have to specify the options we care about (and not have a magic number when we want to skip one of min layer size and default span size)
3) We don't have to update all the call sites if we add additional parameters.

Reply:
That makes sense to me, but I feel it worths a separate PR given that it requires more change and here we are just dedup some functions (and below):

This will be almost a direct copy of [soci.buildConfig](https://github.com/awslabs/soci-snapshotter/blob/792a8996dc02311d9ca85c23fd18ae5563212f28/soci/soci_index.go#L132-L167). Maybe we should just export `buildConfig` and use it in, e.g., `buildIndex`, instead of duplicating these opt functions.

WDYT @Kern--

Reply:
A separate PR is ok, but I think I'd prefer to do the duplication because this models CLI parameters which may be different than the library options. E.g. https://github.com/awslabs/soci-snapshotter/pull/308 adds an `ArtifactDB` to `buildConfig` which doesn't make sense in this context.

Reply:
SGTM.